tests/unit/synapseclient/core/credentials/unit_test_cred_provider.py

```diff
@@ -245,27 +274,40 @@ def teardown(self):
         self.get_config_authentication__patcher.stop()
 
     def test_get_auth_info__user_arg_username_is_None(self):
-        user_login_args = UserLoginArgs(username=None, password=None, api_key=None, skip_cache=False)
+        user_login_args = UserLoginArgs(username=None, password=None, api_key=None, skip_cache=False, auth_token=None)
 
         returned_tuple = self.provider._get_auth_info(self.syn, user_login_args)
 
         assert self.expected_return_tuple == returned_tuple
         self.mock_get_config_authentication.assert_called_once_with()
 
     def test_get_auth_info__user_arg_username_matches_config(self):
-        user_login_args = UserLoginArgs(username=self.username, password=None, api_key=None, skip_cache=False)
+        user_login_args = UserLoginArgs(
+            username=self.username,
+            password=None,
+            api_key=None,
+            skip_cache=False,
+            auth_token=None,
+        )
 
         returned_tuple = self.provider._get_auth_info(self.syn, user_login_args)
 
         assert self.expected_return_tuple == returned_tuple
         self.mock_get_config_authentication.assert_called_once_with()
 
     def test_get_auth_info__user_arg_username_does_not_match_config(self):
```

Comment:
test name does not match intended test. 
I would assume that if a uncached username is being used, then no token should be returned

Reply:
This was also masking a perhaps ambiguous case that is probably not correct. If you had a username and an auth token in your config and supplied a username at the login prompt and the login prompt did not match the config it would allow you to login using the config token. I've removed that ambiguity (i.e. if you supply a username arg and there is a username in the config file it won't use the config file at all). I don't think this is probably a likely case because there's not really any reason for someone to add a username and auth token to their config file now, but it seems better to have no ambiguity.